tests/Integration/WhereConditions/WhereHasConditionsDirectiveTest.php

```diff
@@ -0,0 +1,140 @@
+<?php
+
+namespace Tests\Integration\WhereConditions;
+
+use Nuwave\Lighthouse\WhereConditions\WhereConditionsServiceProvider;
+use Tests\DBTestCase;
+use Tests\Utils\Models\User;
+
+class WhereHasConditionsDirectiveTest extends DBTestCase
+{
+    protected $schema = /** @lang GraphQL */'
+    type User {
+        id: ID!
+        name: String
+        email: String
+    }
+
+    type Post {
+        id: ID!
+        title: String
+        body: String
+    }
+
+    type Company {
+        id: ID!
+        name: String
+    }
+
+    type Query {
+        posts(
+            hasUser: _ @whereHasConditions(relation: "user")
+        ): [Post!]! @all
+
+        users(
+            hasCompany: _ @whereHasConditions(relation: "company")
+            hasPost: _ @whereHasConditions(relation: "posts")
+        ): [User!]! @all
+
+        companies(
+            hasUser: _ @whereHasConditions(relation: "users")
+        ): [Company!]! @all
+
+        whitelistedColumns(
+            hasCompany: _ @whereHasConditions(relation: "company", columns: ["id", "camelCase"])
+        ): [User!]! @all
+
+        withoutRelation(
+            hasCompany: _ @whereHasConditions
+        ): [User!]! @all
+    }
+    ';
+
+    protected function getPackageProviders($app)
+    {
+        return array_merge(
+            parent::getPackageProviders($app),
+            [WhereConditionsServiceProvider::class]
+        );
+    }
+
+    public function testIgnoreEmptyConditions(): void
+    {
+        factory(User::class)->create([
+            'company_id' => null,
+        ]);
+
+        $this->graphQL(/** @lang GraphQL */ '
+        {
+            users(
+                hasCompany: {}
```

Comment:
I think we could differentiate semantically between `null` (don't use this filter at all) and `{}` (filter but without conditions).

Reply:
See [above](https://github.com/nuwave/lighthouse/pull/1140#discussion_r366388616) for my reasoning. I agree that it would be nice to be able to query for relationship existence without a condition though. But it would also be good to have consistent behavior between `@whereConditions` and `@whereHasConditions`.

I could imagine 2 solutions to achieve both:

1. Allow `null` as a value for both `@whereConditions` and `@whereHasConditions`, so that this is the unified value that has no effect
2. Add a separate `@hasRelationships` directive that accepts an array of relationships which are tested through Laravel's [has method](https://laravel.com/docs/5.8/eloquent-relationships#querying-relationship-existence)

I do like the second option more as it is a more explicit and less hacky way to query for relationship existence without a condition, and it could also accept an array of relationships. But I'm not sure if you're okay to add an extra directive just for this purpose.

Reply:
*1.* should already be that case if i am not mistaken.

Having `{}` mean "empty conditions" seems like it is logically consistent and also makes sense from a client perspective.

Imagine you have some kind of UI for the filter and you add the condition that your model has to have a related model that fulfils some complex, nested conditions. To widen the range of results, you remove conditions one by one. At a certain point, you will be left with an empty set `{}` of conditions.

Now, it would be surprising to me if that would cause the relationship restriction to not apply at all now, that should be an additional step: removing the filter altogether by making the argument `null`.

Reply:
We should document and test this behaviour, of course.

Reply:
That sounds reasonable too - I'm fine with solution *1.* as well. Do you want to document and test the `null` vs. `{}` behavior or should I do that?

Good job on the documentation separation btw, this is a nice way to unify the docs for both directives.

Reply:
If you can add that to this PR, that would be nice. You have done a great job so far with this.

Reply:
Okay, done:

- Using `null` on `@whereConditions`: No effect
- Using `{}` on `@whereConditions`: No effect
- Using `null` on `@whereHasConditions`: No effect
- Using `{}` on `@whereHasConditions`: Query for relationship existence without condition